src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -326,6 +327,7 @@ public static void AddCommonServices(IServiceCollection services)
             services.AddSingleton<HostIdValidator>();
             services.AddSingleton<IHostIdProvider, ScriptHostIdProvider>();
             services.TryAddSingleton<IScriptEventManager, ScriptEventManager>();
+            services.AddSingleton<IWorkerProfileManager, WorkerProfileManager>();
```

Comment:
Just as a precaution, checking if this service registration would impact cold start of languages that do not use profiles. 
Bringing this to attention as previously we had decided to avoid service registration. 
@safihamid could you please let us know your opinion on this

Reply:
Oh yes, this will have a cold start impact. we reviewed and measured a similar PR related to this and Fabio's suggestion fixed it. @shreyas-gopalakrishna can you point to that PR please?

Reply:
Initial PR - https://github.com/Azure/azure-functions-host/pull/8365
PR where we removed this - https://github.com/Azure/azure-functions-host/pull/8502

Reply:
@fabiocav and @brettsam  discussed this and we decided this was the best way to unblock the java team for now. In future this will need a full refactor but we don't have the capacity for that right now.

Reply:
I did have a quick sync with @fabiocav about this and he mentioned that this should improve cold start as we are now only registering once at the WebHost scope whereas before it was getting registered multiple times at the WebHost and JobHost scope

Reply:
yes that's right. I see they are being added in AddCommonServices. Didn't notice it initially.